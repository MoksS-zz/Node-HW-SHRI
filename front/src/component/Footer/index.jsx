import React from "react";
import "./Footer.scss";
import "./Link.scss";
import "./Copyright.scss";

function Footer() {
  return (
    <footer className="Footer">
      <div className="Footer_Content">
        <div className="Link">
          <a className="Link_ATag" href="/#">Support</a>
          <a className="Link_ATag" href="/#">Learning</a>
        </div>
        <div className="Copyright">
          <p className="Copyright_Text">Â© 2020 Your Name</p>
        </div>
      </div>
    </footer>
  );
}

export default Footer;