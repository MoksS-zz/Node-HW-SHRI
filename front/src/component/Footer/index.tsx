import React, { FC } from "react";
import "./Footer.scss";
import "./Link.scss";
import "./Copyright.scss";
import store from "../../store";

const currentLang = store.getState().lang;

const Footer: FC = () => {
  const { lang } = window;
  return (
    <footer className="Footer">
      <div className="Footer_Content">
        <div className="Link">
          <a className="Link_ATag" href="/#">{lang.Footer.Support}</a>
          <a className="Link_ATag" href="/#">{lang.Footer.Learning}</a>
          <a className="Link_ATag" 
            href={currentLang === 'ru' ? "/en" : "/ru"}
          >
            {lang.Footer.lang}</a>
        </div>
        <div className="Copyright">
          <p className="Copyright_Text">{lang.Footer.Copyright}</p>
        </div>
      </div>
    </footer>
  );
}

export default Footer;