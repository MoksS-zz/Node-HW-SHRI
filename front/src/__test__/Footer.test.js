import React from "react";
import renderer from "react-test-renderer";
import Footer from "../component/Footer";

it("renders Footer correctly", () => {
  const footer = renderer
    .create(<Footer />)
    .toJSON();

  expect(footer).toMatchSnapshot();
});
