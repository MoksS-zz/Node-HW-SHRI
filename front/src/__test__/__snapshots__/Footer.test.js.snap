// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<footer
  className="Footer"
>
  <div
    className="Footer_Content"
  >
    <div
      className="Link"
    >
      <a
        className="Link_ATag"
        href="/#"
      >
        Support
      </a>
      <a
        className="Link_ATag"
        href="/#"
      >
        Learning
      </a>
    </div>
    <div
      className="Copyright"
    >
      <p
        className="Copyright_Text"
      >
        Â© 2020 Your Name
      </p>
    </div>
  </div>
</footer>
`;
